src/Features/CSharp/Portable/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionDiagnosticAnalyzer.cs

```diff
@@ -61,9 +61,10 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
                 return;
             }
 
-            context.ReportDiagnostic(Diagnostic.Create(Descriptor,
+            context.ReportDiagnostic(DiagnosticHelper.Create(Descriptor,
```

Comment:
if this is an optional arg, we should consider making mandatory, so that this can't happen.

Reply:
This is a mandatory argument already for `DiagnosticHelper.Create` API. The issue was the analyzer directly invoked the `Diagnostic.Create` public API, which is meant for all non-code style analyzers.

Reply:
ah, got it.  yeah... hrmm, i wonder if we should have an analyzer to catch usages of Diagnostic.Create!